File: Excel/487 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/487 Maximum Frequency Characters.xlsx"
input = read_xlsx(path, range = "A2:A11")
test  = read_xlsx(path, range = "B2:C11")

find_most_freq_char = function(input) {
  result = input %>% 
    strsplit("") %>% 
    unlist() %>% 
    table()
  df = data.frame(
    Characters = names(result),
    Frequency = as.numeric(result)) %>% 
    filter(Frequency == max(Frequency)) %>%
    summarise(Characters = paste(sort(Characters), collapse = ", "), .by = Frequency) %>%
    select(Characters, Frequency)
  return(df)
}

result = input$Strings %>%
  map(find_most_freq_char) %>%
  bind_rows()

# Validation on eye, all correct. In three cases there is different sorting than
#  in provided example but with the same characters.

File: Excel/400-499/487/487 Challenge.py
import pandas as pd

path = "487 Maximum Frequency Characters.xlsx"

input = pd.read_excel(path, usecols="A", skiprows=1)
test  = pd.read_excel(path, usecols="B:C", skiprows=1)

def max_freq_char(s):
    s = list(s)
    freq = {}
    [freq.update({c: freq.get(c, 0) + 1}) for c in s]
    max_freq = max(freq.values())
    max_freq_chars = ", ".join([k for k, v in freq.items() if v == max_freq])
    return pd.DataFrame({"Characters": [max_freq_chars], "Frequency": [max_freq]})

output = pd.concat([max_freq_char(s) for s in input["Strings"]], ignore_index=True)

# Validation on eye. Not every result has the same order of characters as in the test,
# but pointed characters and their frequencies are correct.